Source/Scene/ImplicitTileCoordinates.js

```diff
@@ -178,12 +219,322 @@ ImplicitTileCoordinates.prototype.deriveChildCoordinates = function (
   // Quadtree
   return new ImplicitTileCoordinates({
     subdivisionScheme: this.subdivisionScheme,
+    subtreeLevels: this.subtreeLevels,
     level: level,
     x: x,
     y: y,
   });
 };
 
+/**
+ * Given the (level, x, y, [z]) coordinates of the parent and
+ * a (level, x, y, [z]) relative offset, compute the coordinates of the descendant.
+ *
+ * @param {ImplicitTileCoordinates} localCoordinates The local coordinates of the descendant
+ * @returns {ImplicitTileCoordinates} The tile coordinates of the descendant
+ * @private
+ */
+ImplicitTileCoordinates.prototype.deriveDescendantCoordinates = function (
+  localCoordinates
+) {
+  var subdivisionScheme = this.subdivisionScheme;
+  var subtreeLevels = this.subtreeLevels;
+  var tileLevel = this.level;
+  var tileX = this.x;
+  var tileY = this.y;
+
+  //>>includeStart('debug', pragmas.debug);
+  Check.typeOf.object("localCoordinates", localCoordinates);
+  if (subdivisionScheme !== localCoordinates.subdivisionScheme) {
+    throw new DeveloperError("coordinates must have same subdivisionScheme");
+  }
+
+  if (subtreeLevels !== localCoordinates.subtreeLevels) {
+    throw new DeveloperError("coordinates must have same subtreeLevels");
+  }
+  //>>includeEnd('debug');
+
+  var globalLevel = tileLevel + localCoordinates.level;
+  var globalX = (tileX << localCoordinates.level) + localCoordinates.x;
+  var globalY = (tileY << localCoordinates.level) + localCoordinates.y;
+
+  if (subdivisionScheme === ImplicitSubdivisionScheme.OCTREE) {
+    var tileZ = this.z;
+    var globalZ = (tileZ << localCoordinates.level) + localCoordinates.z;
+
+    return new ImplicitTileCoordinates({
+      subdivisionScheme: subdivisionScheme,
+      subtreeLevels: subtreeLevels,
+      level: globalLevel,
+      x: globalX,
+      y: globalY,
+      z: globalZ,
+    });
+  }
+
+  // Quadtree
+  return new ImplicitTileCoordinates({
+    subdivisionScheme: subdivisionScheme,
+    subtreeLevels: subtreeLevels,
+    level: globalLevel,
+    x: globalX,
+    y: globalY,
+  });
+};
+
+/**
+ * Get the coordinates of the subtree that contains this tile
```

Comment:
Could we make sure that in the documentation of these functions that we're clear what are local and what are global coordinates?